platforms/documentation/docs/src/docs/userguide/reference/dsl-apis/public_apis.adoc

```diff
@@ -18,192 +18,68 @@
 
 Gradle provides many public APIs that allow developers to interact with various build system components.
 
-Any class name containing `Internal`, or that is located in a package containing the segment `.internal.` is by definition not part of the public Gradle API and should not be relied upon for stable usage.
-
-The following table summarizes public Gradle APIs and their common use cases.
-This is *not* an exhaustive list:
-
-[cols="1,1,2"]
+Any class or package name containing `Internal`, or located in a package segment `.internal.`, is not part of the public Gradle API and should not be relied upon for stable usage.
+Use of Gradle internal APIs in plugins and build scripts can break builds when either Gradle or plugins change.
+
+The list of public API packages is maintained in the link:https://github.com/gradle/gradle/blob/master/buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/public-api.kt[Gradle public API definition] (excluding any subpackages containing internal).
+It is also available below:
+
+[source,text]
+----
+org/gradle/*
+org/gradle/api/**
+org/gradle/authentication/**
+org/gradle/build/**
+org/gradle/buildconfiguration/**
+org/gradle/buildinit/**
+org/gradle/caching/**
+org/gradle/concurrent/**
+org/gradle/deployment/**
+org/gradle/external/javadoc/**
+org/gradle/ide/**
+org/gradle/ivy/**
+org/gradle/jvm/**
+org/gradle/language/**
+org/gradle/maven/**
+org/gradle/nativeplatform/**
+org/gradle/normalization/**
+org/gradle/platform/**
+org/gradle/plugin/devel/**
+org/gradle/plugin/use/*
+org/gradle/plugin/management/*
+org/gradle/plugins/**
+org/gradle/process/**
+org/gradle/testfixtures/**
+org/gradle/testing/jacoco/**
+org/gradle/tooling/**
+org/gradle/swiftpm/**
+org/gradle/model/**
+org/gradle/testkit/**
+org/gradle/testing/**
+org/gradle/vcs/**
+org/gradle/work/**
+org/gradle/workers/**
+org/gradle/util/**
+----
+
+== Avoiding internal APIs
```

Comment:
Is there a reason you added this section? I personally am not aware of widespread usage of these particular APIs in the wild, and the `util` package is actually public API that we don't discourage. The internal parts are in `.internal.` already.

Reply:
It was removed from Grade 7 and put it back. 
Removed. TY.